plugins/source/aws/resources/services/elasticsearch/vpc_endpoints_fetch.go

```diff
@@ -0,0 +1,57 @@
+package elasticsearch
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/elasticsearchservice"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client/services"
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+func fetchElasticsearchVpcEndpoints(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
+	svc := meta.(*client.Client).Services().Elasticsearchservice
+
+	listInput := new(elasticsearchservice.ListVpcEndpointsInput)
+	var vpcEndpointIDs []string
+	// get the IDs first
+	for {
+		out, err := svc.ListVpcEndpoints(ctx, listInput)
+		if err != nil {
+			return err
+		}
+
+		for _, summary := range out.VpcEndpointSummaryList {
+			vpcEndpointIDs = append(vpcEndpointIDs, *summary.VpcEndpointId)
+		}
+
+		if out.NextToken == nil {
+			break
+		}
+
+		listInput.NextToken = out.NextToken
+	}
+
+	return describeVPCEndpoints(ctx, svc, res, vpcEndpointIDs)
+}
+
+func describeVPCEndpoints(ctx context.Context, svc services.ElasticsearchserviceClient, res chan<- any, vpcEndpointIDs []string) error {
+	const maxLen = 100
+	if l := len(vpcEndpointIDs); l > maxLen {
+		if err := describeVPCEndpoints(ctx, svc, res, vpcEndpointIDs[:l/2]); err != nil {
+			return err
+		}
+		return describeVPCEndpoints(ctx, svc, res, vpcEndpointIDs[l/2:])
+	}
```

Comment:
Why is this doing division by 2? It's unlike other resolvers, so I think it deserves a comment in the code, at least

Reply:
they support amounts from 5 to 100, so we want to utilize as little calls as possible

Reply:
I see - I could be missing something, but I think it's unnecessary to reach for recursion here. Couldn't we call `DescribeVpcEndpoints` N / 100 times with the max number, and once with the remaining items? That would still be optimal in terms of the number of calls.

Reply:
We want to make as little API calls as possible, so we want to pass as many IDs to the describe calls as we can

Reply:
as for the `/100` - the minimum amount stated is 5 (IDK why, maybe they fixed it), so I'd better err on the safe side

Reply:
Ohh, I get it now, there is a _minimum_ number of items as well 😅 I missed that before, thanks!

Reply:
I guess it could still be done without recursion in a variety of ways, but I understand why the code is like it is now, so I won't press the point :)

Reply:
rewriting for now, if we hit issues with AWS API -- we'll reconsider